lib/RGBot.js

```diff
@@ -294,83 +593,153 @@ const RGBot = class {
      */
     async handlePath(pathFunc, options = {}) {
         const interval = options.interval || 5000;
-        let previousPosition = this.#bot.entity.position;
-        let wasActive = true;
+        let previousPosition = this.position();
         let stuck = false;
 
         const checkPosition = () => {
-            let currentPosition = this.#bot.entity.position;
-            let isActive = this.#bot.pathfinder.isMining() || this.#bot.pathfinder.isBuilding();
-            if (currentPosition.equals(previousPosition, 0.005) && !wasActive && !isActive) {
-                // if the bot hasn't moved or performed other actions then we are stuck
-                // stop pathfinder and remove its current goal
-                console.error(`handlePath: Bot is stuck. Stopping current path.`);
-                stuck = true;
-                this.#bot.pathfinder.setGoal(null);
-                this.#bot.pathfinder.stop();
-            } else {
-                previousPosition = currentPosition;
-                wasActive = isActive;
+            try {
+                let currentPosition = this.#bot.entity.position;
+                if (currentPosition.distanceTo(previousPosition) < 2 && !this.#bot.pathfinder.isMining() && !this.isLookingInContainer && !this.isCrafting) {
+                    // if the bot hasn't moved or performed other actions then we are stuck
+                    // stop pathfinder and remove its current goal
+                    this.#log(`!!!!!!!!!!!!!! Bot is stuck at ${this.vecToString(currentPosition)}. Stopping pathfinding...`);
```

Comment:
Hmm I wouldn't put something about "you should go hit your bot" as a suggestion because someone should really be able to use the result from handlePath (did I make it to my destination?) to programmatically determine what action the bot needs to take next.

I'm finding it hard to come up with a solid suggestion because it really depends on why your bot is stuck and what you were trying to have them do. As Zack said, sometimes the bot gets physically stuck inside a block that it'll need to break to free itself. But sometimes the pathfinding just does wonky things... for example, my poppy bot kept getting stuck like clockwork on one block in the starting village, a one-block-tall ledge that is a short distance from one of the houses. There's a poppy underneath underneath that house's roof that the bot needed to collect but it would just stand at the edge of the ledge and stop moving. I eventually found out that if I broke the roof block above the poppy, the bot magically started navigating again. So it wasn't physically stuck, it just couldn't figure out how to get to my goal.

Reply:
It seems like there are a lot of reasons why a bot would get stuck, so maybe let's say "please visit http://localhost:3000/documentation/faq#my-bot-gets-stuck-while-pathfinding" (that is a real link I just added to our docs